sos/plugins/kubernetes.py

```diff
@@ -33,12 +37,7 @@ class kubernetes(Plugin, RedHatPlugin):
     ]
 
     def check_is_master(self):
-        if any([
-            path.exists("/var/run/kubernetes/apiserver.key"),
-            path.exists("/etc/origin/master/master-config.yaml")
-        ]):
-            return True
-        return False
+        return any([path.exists(f) for f in self.files])
```

Comment:
The logic here looks a bit weird to me: we have a plugin files list, which, if any of the files exist causes the plugin to run (the plugin will also run if those files do not exist, but one or more package matches are present).

Then we have a separate `check_is_master()` method, that *used* to check two specific "master" paths (the API server key, and the master-config YML file).

Now it just does the same as the default `check_enabled()` does for the `files` list: it returns `True` if any of the `Kubernetes.files` paths exist - so if the plugin is running (at least from a file trigger), then `check_is_master()` is always `True`.

Is there something subtle I've missed here? If the logic is correct then we should be able to dispense with all the call to `check_is_master()`; if it's not then it seems like there's some false sharing between the plugin `files` list and the list of files that `check_is_master()` should be operating on.

Reply:
Right, the `files` list is strictly locations that a kubernetes master would have that a worker node would not have - there are no files (at least that I'm aware of) that a node would have that a master wouldn't have. For worker node collection we're relying on just the package check.

The reason for this is that we still want to collect the `kube*` journals but literally everything else in the plugin would not be available on the worker nodes.